decorators.py

```diff
@@ -16,16 +16,24 @@ def decorated_function(*args, **kwargs):
     return decorated_function
 
 
-def check_access_level(access_level):
+def check_access_level(*allowed_roles):
     def decorator(f):
         @wraps(f)
         def decorated_function(*args, **kwargs):
+            user_id_profile = kwargs.get('user_id')
```

Comment:
```suggestion
```

You'd better do the check in the appropriate route processing function so that `user_id` has the right meaning. Here you will only check for roles.